inginious/frontend/pages/preferences/profile.py

```diff
@@ -54,20 +56,18 @@ def save_profile(self, userdata, data):
             msg = _("Passwords don't match !")
             return result, msg, error
         elif self.app.allow_registration and len(data["passwd"]) >= 6:
-            oldpasswd_hash = UserManager.hash_password(data["oldpasswd"])
-            passwd_hash = UserManager.hash_password(data["passwd"])
 
-            match = {"username": self.user_manager.session_username()}
             if "password" in userdata:
-                match["password"] = oldpasswd_hash
-
-            result = self.database.users.find_one_and_update(match,
-                                                             {"$set": {"password": passwd_hash}},
-                                                             return_document=ReturnDocument.AFTER)
-            if not result:
-                error = True
-                msg = _("Incorrect old password.")
-                return result, msg, error
+                user = self.user_manager.auth_user(self.user_manager.session_username(), data["oldpasswd"], False)
```

Comment:
I'm a little bit confused with prior code already, but can you confirm that a user with no password (registered through social networks for instance) can actually set a password to use password authentication ?

Reply:
They can now. I changed the structure for handling user password change in this PR. Tell me if your ok with it.